plugins/languages/otter.nix

```
@@ -0,0 +1,58 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "otter";
+  originalName = "otter.nvim";
+  defaultPackage = pkgs.vimPlugins.otter-nvim;
+
+  maintainers = [ lib.maintainers.perchun ];
+
+  settingsOptions = {
+    lsp = {
+      hover = {
+        border = helpers.defaultNullOpts.mkListOf lib.types.str [
+          "╭"
+          "─"
+          "╮"
+          "│"
+          "╯"
+          "─"
+          "╰"
+          "│"
+        ] "";
+      };
+      diagnostic_update_events = helpers.defaultNullOpts.mkListOf lib.types.str [ "BufWritePost" ] ''
+        `:h events` that cause the diagnostics to update. Set to:
+        [ "BufWritePost" "InsertLeave" "TextChanged" ] for less performant
+        but more instant diagnostic updates
+      '';
+    };
+    buffers = {
+      set_filetype = helpers.defaultNullOpts.mkBool false ''
+        if set to true, the filetype of the otterbuffers will be set.
+        otherwise only the autocommand of lspconfig that attaches
+        the language server will be executed without setting the filetype
+      '';
+      write_to_disk = helpers.defaultNullOpts.mkBool false ''
+        write <path>.otter.<embedded language extension> files
+        to disk on save of main buffer.
+        useful for some linters that require actual files
+        otter files are deleted on quit or main buffer close
```

```suggestion
        Useful for some linters that require actual files
        otter files are deleted on quit or main buffer close.
```

Generally, try to capitalize & punctuate correctly (I know this can be painful when copying upstream docs)